packages/webdriverio/src/commands/element/addValue.ts

```diff
@@ -38,23 +22,24 @@ const isValidType = (value: unknown) => (
  * </example>
  *
  * @alias element.addValue
- * @param {string | number | Array<string | number>}        value                       value to be added
- * @param {CommandOptions=}                                 options                     command options (optional)
- * @param {boolean}                                         options.translateToUnicode  enable translation string to unicode value automatically
+ * @param {string | number}  value  value to be added
  *
  */
 export default function addValue (
     this: WebdriverIO.Element,
-    value: Value | Value[],
-    { translateToUnicode = true }: CommandOptions = {}
+    value: string | number
 ) {
-    if (!isValidType(value)) {
-        log.warn('@deprecated: support for type "string", "number" or "Array<string | number>" is deprecated')
-    }
-
-    if (!this.isW3C) {
-        return this.elementSendKeys(this.elementId, transformToCharString(value, translateToUnicode) as any as string)
+    /**
+     * The JSONWireProtocol allowed array values and use of special characters when adding a value to an input.
+     * With the W3C protocol this was not possible anymore. This is a type check to ensure users are aware of
+     * this transition.
+     */
+    if (Array.isArray(value) || !value || typeof value === 'boolean') {
+        throw new Error(
```

Comment:
How do you feel about using only one type (string)?

I have personally never liked using anything else since it feels odd to throw anything but a string into an input element. It's not a strong opinion but I'm curious if you feel the same.

Reply:
No strong opinion either but I tend towards allowing `number` as well because it simplifies usage when an number is being passed in e.g. from parsing a CSV file or whatever.